app.js

```diff
@@ -0,0 +1,468 @@
+ *               - If databaseResponse is null the result was not inserted in
+ *                 any graph, most likely because file IRI was not in any graph.
+ * @return [400] if request malformed.
+ * @return [422] if no related physical file is found in database.
+ */
+app.post(
+  '/scan',
+  bodyParser.json({ limit: '50mb' }),
+  async function (req, res) {
+    try {
+      const body = req.body;
+      if (LOG_INCOMING_SCAN_REQUESTS) {
+        console.log(`Receiving scan request : ${JSON.stringify(body)}`);
+      }
+
+      const logicalFileIRI = body.file;
+      if (
+        !(
+          typeof logicalFileIRI === 'string' || logicalFileIRI instanceof String
+        ) ||
+        !logicalFileIRI.length
+      ) {
+        return res.status(400).send('`file` not a non-empty String');
+      }
+
+      if (logicalFileIRI.slice(0, 8) === 'share://') {
+        // TODO: Be flexible and lookup the logical file IRI? Can we assume
+        //       that, even if the physical file IRI exists in multiple graphs,
+        //       they will all be related to the same logical file IRI?
+        return res
+          .status(422)
+          .send('`file` is a physical file IRI, should be a logical file IRI');
+      }
+
+      // TODO: Check for existence of `<logicalFileIRI> a nfo:FileDataObject`?
+
+      const scanFileResult = await scanFile(logicalFileIRI);
+
+      const storeResult = await storeMalwareAnalysis(
+        logicalFileIRI,
+        scanFileResult.stixMalwareAnalysis,
+      );
+
+      console.dir(
+        {
+          file: logicalFileIRI,
+          ...scanFileResult,
+          ...storeResult,
+        },
+        { depth: null },
+      );
+
+      res.status(201).send(storeResult.resourceObject);
+    } catch (e) {
+      console.log(e);
+      res.status(500).send('Uncaught error in /scan: ' + e);
+      // TODO: Same question as for /delta: Re-throw error?
+    }
+  },
+);
+
+app.use(errorHandler);
+
+/**
+ * Scans a file for viruses.
+ *
+ * @async
+ * @function
+ * @param {String} fileIRI - IRI file to scan. This can be a logical/virtual
+ *                           file IRI or a physical/stored file IRI.
+ * @returns {Object} Properties:
+ *          .stixMalwareAnalysis - The malware analysis details. Remarks:
+ *                                 - result: If "unknown", see .error.
+ *                                 - sampleRef: Not set, see .lookups.
+ *          .error - Error object (if any).
+ *          .lookups - Results of lookups from the requested file IRI to
+ *                     the physical file path.
+ */
+async function scanFile(fileIRI) {
+  const ret = {
+    stixMalwareAnalysis: {
+      analysisStarted: new Date(),
+      analysisEnded: undefined,
+      result: 'unknown',
+      resultName: undefined,
+    },
+    error: undefined,
+    lookups: undefined,
+  };
+  let physicalFileIRI;
+  let file;
+
+  try {
+    physicalFileIRI =
+      fileIRI.slice(0, 8) === 'share://'
+        ? fileIRI
```

Comment:
Since I didn't see it being used on physical files, i would scrap this to keep code a bit simpler